### PR TITLE
Fix BooleanInput props

### LINKED ISSUE #7550: <BooleanInput> - cannot set custom icons
In v3 I had customized <BooleanInput> to use specific icons/colors, like the manual says it should be done:

`import { BooleanInput } from 'react-admin';
import FavoriteIcon from '@mui/icons-material/Favorite';

<BooleanInput
    source="favorite"
    options={{
        checkedIcon: <FavoriteIcon />,
    }}
/>`

It worked as expected in v3.

In v4, it does not work anymore.

Looking at the source of BooleanInput.tsx, it looks like the 'options' is not passed on to the underlying <Switch> component. And that 'options' is removed by the call to sanitizeInputRestProps() ?

### PR DESCRIPTION
**Pull request by WiXSL**
Fixes #7550

### CODE REVIEW

File: packages/ra-ui-materialui/src/input/BooleanInput.tsx [hunk 1]
```diff
@@ -62,7 +63,7 @@ export const BooleanInput = (props: BooleanInputProps) => {
     return (
         <FormGroup
             className={clsx('ra-input', `ra-input-${source}`, className)}
-            {...sanitizeInputRestProps(rest)}
+            row={row}
```
Comment (WiXSL) [inline packages/ra-ui-materialui/src/input/BooleanInput.tsx:66]: `row` and `sx` are really the only props to pass to `FormGroup` component.
And I'm not sure about setting `sx` to both `FormGroup` and `Switch`, so I choose the latter

**Review by slax57 — APPROVED** [review]

File: docs/Upgrade.md [hunk 2]
```diff
@@ -2636,6 +2636,21 @@ import { BooleanInput } from 'react-admin';
 />
 ```
 
+### `BooleanInput` `options` Prop Has Been Removed
```
Comment (fzaninotto) [inline docs/Upgrade.md:2639]: It's a bit too late to introduce breaking change. So either we wait for 5.0, or we do it in a backwards compatible way.

File: packages/ra-ui-materialui/src/input/BooleanInput.tsx (hunk 1 above)
Comment (fzaninotto) [reply packages/ra-ui-materialui/src/input/BooleanInput.tsx:66]: I think you should put the `sx` on the FormGroup. That way, the developer can customize both the FormGroup and the Switch styles

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: docs/BooleanInput.md [hunk 3]
```diff
@@ -24,12 +24,7 @@ You can use the `options` prop to pass any option supported by the MUI's `Switch
 import { BooleanInput } from 'react-admin';
 import FavoriteIcon from '@mui/icons-material/Favorite';
 
-<BooleanInput
-    source="favorite"
-    options={{
-        checkedIcon: <FavoriteIcon />,
-    }}
-/>
+<BooleanInput source="favorite" checkedIcon={<FavoriteIcon />} />
```
[the remark this answers is not in the document]
Comment (WiXSL) [reply docs/BooleanInput.md:27]: No, It's perfectly fine, I have to add the `options` prop for backward compatibility. The definition of `BooleanInputProps` includes `SwitchProps`

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**